### ISSUE TITLE
Native toolchain with C++ fails to build from vanilla GCC 11.2

### ISSUE THREAD

**Issue by abrodkin** [foss-for-synopsys-dwc-arc-processors/toolchain#414]
Just bumped into this problem [https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100017](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100017) trying to build "native" toolchain from vanilla GCC 11.2 with C++ enabled:
```
[INFO ]  Installing final gcc compiler for build: done in 204.68s (at 15:55)
[INFO ]  =================================================================
[INFO ]  Installing final gcc compiler
[EXTRA]    Configuring final gcc compiler
[EXTRA]    Building libiberty
[EXTRA]    Building final gcc compiler
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:58:11: error: 'fenv_t' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:59:11: error: 'fexcept_t' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:62:11: error: 'feclearexcept' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:63:11: error: 'fegetexceptflag' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:64:11: error: 'feraiseexcept' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:65:11: error: 'fesetexceptflag' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:66:11: error: 'fetestexcept' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:68:11: error: 'fegetround' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:69:11: error: 'fesetround' has not been declared in '::'
[ERROR]    .../.build/HOST-arc-snps-linux-gnu/arc-snps-linux-gnu/build/build-cc-gcc-final/arc-snps-linux-gnu/libstdc++-v3/include/fenv.h:71:11: error: 'fegetenv' has not been declared in '::'
```

There's no final solution in the upstream bugzilla with a couple of suggestions which might help, but we'll need check on that once we start building RC's for `arc-2021.09`.

**Comment by abrodkin** [comment]
With CT-NG this could be worked around with `CT_TARGET_CFLAGS="-nostdinc++"`.

**Comment by abrodkin** [comment]
Also temporary fixed with https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/commit/059e632e904d91a522cd359c6949654d8de16780 in `arc-2021.09` for the release in a humble hope it finally gets fixed upstream by the time of GCC 11.3 release.

**Comment by abrodkin** [comment]
Work-around posted and accepted in upstream Crosstool-NG (https://github.com/crosstool-ng/crosstool-ng/commit/9ddd4b7883a9f7b341bedfa5d4e7a96446140f1b) & in our fork, see https://github.com/foss-for-synopsys-dwc-arc-processors/crosstool-ng/commit/9ddd4b7883a9f7b341bedfa5d4e7a96446140f1b.
